### ISSUE TITLE
Why are groups still used in `tags.py`

### ISSUE THREAD

**Issue by Ibrahim2750mi** [python-discord/bot#2423]
According to these [lines](https://github.com/python-discord/bot/blob/main/bot/exts/info/tags.py#L150-#L151)
```python
# Files directly under `base_path` have an empty string as the parent directory name
tag_group = parent_dir.name or None
```
`tag_group` will be None until the tag is in a sub-directory inside `bot/resources/tags/` eg: `bot/resources/tags/fstring/fstring.md` and currently we don't have sub-directories in the folder.

Moving them to sub-directories might also solve #2345 as then the aliases will be listed as bullets under the directory name eg: 
![2023-03-02_03-41](https://user-images.githubusercontent.com/74553450/222276833-d2d447e0-af12-4876-a789-d14175c21a85.png)
NOTE: I have kept the directories name as same as the .md(s) here, it could look a lot better if used correct names in parent bullet.

My suggestion is either we nuke them or sort the tags in the directory to use the group functionality of the code.

**Comment by janine9vn** [comment]
The intent was to sort tags into groups and sub-directories but no one has taken it up yet. I'm firmly against removing this feature just because we haven't used it yet.

For instance, all the d.py related tags can go into a subdirectory, it just requires someone making that PR

**Comment by Ibrahim2750mi** [comment]
This is the reason I made the issue(if there already an issue for this, then I am sorry), I could sort them up in the sub-directories. 

But first I think we need to finalise the groups we are going to have. I am thinking something along these lines.

+ python-strings
Tags related to strings `fstring`, `str-join` etc.
+ python-discord
Tags related to us like `codeblock`, `ytdl`, `xyproblem` etc.
+ discord.py
Tags related to d.py
+ A-vs-B (can't seem to figure out a better name for this but we have a lot of tags with this format)
The tags with titles like A vs B, `positional-keyword`, `param-arg` etc.
+ python-gotchas
The common python gotchas like `if-name-main`, `or-gotcha` etc.
+ python-beginner-syntax
All the beginner tags like `comparison`, `return`, `self` etc.
+ python-advance
Stuff like `dunder-methods`, `list-comprehensions` etc.
+ python-environment
Stuff outside python code like `pep-8`, `microsoft-build-tools` etc.
+ python-errors
Tags related to errors like `nomudle`, `empty-json` etc.

I feel like there can be more categories added to lighten up `python-beginner-syntax`. Help to suggest group names is appreciated.

**Comment by janine9vn** [comment]
I don't think we need to create groups for every single tag or for most tags. Any additional grouping we add means you need an additional term to invoke it. With your proposal, calling the pep8 tag would be `!python-environment pep-8`. So I want to really avoid grouping things just to group them, there should be a natural purpose to it besides "we're not using this code feature yet." The purpose behind groups was to avoid cluttering the namespace and avoid conflicts around calling tags, so that should still be kept in mind.

For me, the group names that make sense are:
- dpy
- python environments, although I would want to add a shortened alias for this groupname
- server (for tags that are just about our server or other discord servers like faq, guilds, site, tools). This one we'll have to be careful with because if we put any commonly invoked ones under here, it's going to be annoying for regular users that they can't direct invoke anymore.

Also, I don't believe this solves the aliases being listed issue. It just sorts the duplicated names for tags, it doesn't reduce the total clutter or duplicate-nature of it. 

There is an existing issue for utilizing the tag groups as well for some specific tags:
https://github.com/python-discord/bot/issues/1835

Side note: it seems the slash command for tags wasn't tested with tag groups, because the functionality of tag groups has been changed and is no longer really working. It was working when I was testing it with the alias feature though.

**Comment by Ibrahim2750mi** [comment]
Can't we do something like we only require the group name for triggering some specific group tags, an example would be dpy and use the other groups to list the tags in the embed more cleanly? We can check if the `tag.group` is an enforced(enforced meaning it would require the user to add the group name before the tag to invoke it) group if not then it can be triggered by both `!group tag` and `!tag`? For dpy we can make it an enforced group.

**Comment by Ibrahim2750mi** [comment]
I think we are on the same page now.

**Comment by RohanJnr** [comment]
Currently, the new `/tag` command has the syntax: `/tag name:<group_name> <tag_name>`. To use a tag from a group, group name has to be mentioned. This could be quite unintuitive to always enter the group name for a simple tag search. Incase if we are enforcing group name, lets try to keep them as simple as possible and short as possible.
Another approach would be to invoke a tag regardless of its group. Incase there are tags with the same name in multiple groups, the user will be prompted to choose from one group (using buttons here would be nice).

**Comment by HassanAbouelela** [comment]
As Janine pointed out, the tag functionality was greatly reduced with the [slash command migration](https://github.com/python-discord/bot/commit/679cb77a5ad1eefa8c027f3f318d3fb3466cf74d). I’m guessing slash commands don’t allow you to enter invalid commands? That is in part how we were handling tag matching before slash commands (its in the error handler).

If we can’t reimplement this flexibility with the slash-commands, then I don’t think this is a good use case for them, and we might be better off reverting these changes. Slash command benefits don’t outweigh the flexibility.

**Comment by RohanJnr** [comment]
> I’m guessing slash commands don’t allow you to enter invalid commands?
slash commands are kinda gimmicky, like you have to explicitely tell discord that you are providing a value for the `name` argument and only then the error handling takes place
![image](https://github.com/python-discord/bot/assets/37217450/8ac4ceb2-c8e8-4a10-b5eb-ddd4efc2b148)

In the case below, discord correctly parses the name argument
![image](https://github.com/python-discord/bot/assets/37217450/25adf67c-11c2-4008-a8c6-215f3d81594f)


sometimes it just doesn't
![image](https://github.com/python-discord/bot/assets/37217450/8310fd2e-cf43-4020-8738-36044264292c)
Where discord is unable to parse the input as arguments
